### ISSUE TITLE
Sync times have regressed significantly since Ignition launch

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-x-ios#2062]
### Steps to reproduce

1. Foreground the app in a big account
2. Stare at the sync spinner for 4+ seconds
3. Feel very sad that previously it was syncing in ~1s.

Kegan has dug into a rageshake, and is seeing an inexplicable 3s gap in clientside logs (see rageshake for details).  This behaviour started at some point in the last month or so - perhaps when backup got added?

The regression hasn't been picked up by the perftest jig.

#### What did you expect?

Instant sync

#### What happened instead?

Sync regressing.

### Application version

433

### Will you send logs?

Yes

**Comment by ara4n** [comment]
![image](https://github.com/vector-im/element-x-ios/assets/1294269/84de907b-520b-4add-8c33-779ac78138b5)

**Comment by stefanceriu** [comment]
I followed this up on the rageshake and I inspected startup network traffic but there's nothing that points at anything having slowed down client/sdk side. There was a problem in a previous key backup implementation re startup request that but that has since been fixed. 
I will close this until we get more data to work with.

**Comment by ara4n** [comment]
so where is the slowdown coming from in the rageshake? i'm seeing it reliably on every launch.

**Comment by stefanceriu** [comment]
> so where is the slowdown coming from in the rageshake

There were 2 similar rageshakes, I commented on the other one here https://github.com/vector-im/element-x-ios-rageshakes/issues/1074#issuecomment-1810069242

tl;dr; app starts sync -> 2 seconds later sync comes back with **594kiB**(!) of data -> takes 1 second to process rust side, app finally gets it

**Comment by ara4n** [comment]
this is still markedly slower than in sept

**Comment by Avi0n** [comment]
Seeing 4+ sec here as well. Seems to happen when the app has been in the background for a while. Submitted logs referencing this issue.

**Comment by ara4n** [comment]
this has totally changed with the move to SSS. it’s still not as fast as it could be, but it’s faster than it was. https://github.com/element-hq/synapse/issues/17722 might also contribute to perf problems.

keeping this open as it’s still way slower than the original launch.
